vql/server/humio/humio.go

```diff
@@ -0,0 +1,707 @@
+	for name, mappedName := range self.tagMap {
+		var value interface{}
+		value, ok := row.Get(name)
+
+		if ok {
+			payload.Tags[mappedName] = value
+		}
+	}
+}
+
+// We don't want to parse the timestamp.  We don't actually care.  Humio can
+// handle the parsing.
+func (self *HumioQueue) addTimestamp(row *ordereddict.Dict, payload *HumioPayload) {
+	timestamp, ok := row.Get("Time")
+	timezone := ""
+	if !ok {
+		timestamp, ok = row.Get("timestamp")
+	}
+	if !ok {
+		timestamp, ok = row.Get("_ts")
+	}
+	if !ok {
+		timestamp = time.Now().UTC()
+	}
+
+	_, isint := timestamp.(uint64)
+	if isint {
+		timezone = "UTC"
+	}
+
+	payload.Events[0].Timestamp = timestamp
+	payload.Events[0].Timezone = timezone
+}
+
+func NewHumioPayload(row *ordereddict.Dict) *HumioPayload {
+	return  &HumioPayload{
+		Events: []HumioEvent {
+			HumioEvent{
+				Attributes: row,
+			},
+		},
+		Tags: map[string]interface{}{},
+	}
+}
+
+func (self *HumioQueue) rowToPayload(ctx context.Context, row *ordereddict.Dict) *HumioPayload {
+	payload := NewHumioPayload(row)
+
+	self.addClientInfo(ctx, row, payload)
+	self.addMappedTags(row, payload)
+	self.addTimestamp(row, payload)
+
+	return payload
+}
+
+func (self *HumioQueue) postBytes(ctx context.Context, scope vfilter.Scope,
+				  data []byte, count int) (error, bool) {
+	req, err := http.NewRequestWithContext(ctx, "POST", self.endpointUrl,
+					       bytes.NewReader(data))
+	req.Header.Add("User-Agent", constants.USER_AGENT)
+	req.Header.Add("Accept", "application/json")
+	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", self.authToken))
+
+	resp, err := self.httpClient.Do(req)
+	if err != nil && err != io.EOF {
+		self.Log(scope, "request failed: %s", err)
+		return err, true
+	}
+	defer resp.Body.Close()
+
+	self.Debug(scope, "sent %d events %d bytes, response with status: %v",
+		   count, len(data), resp.Status)
+	body := &bytes.Buffer{}
+	_, err = utils.Copy(ctx, body, resp.Body)
+	if err != nil {
+		self.Log(scope, "copy of response failed: %v, %v", resp.Status, err)
+		return err, false
+	}
+
+	if resp.StatusCode != 200 {
+		self.Log(scope, "request failed: %v, %v", resp.Status, body)
+		retry := true
+		// In general these will require restarting the artifact, which will
+		// lose the events anyway.
+		if resp.StatusCode >= 400 && resp.StatusCode < 500 {
+			retry = false
+		}
+		return errHttpClientError{
+			StatusCode: resp.StatusCode,
+			Status: resp.Status,
+			}, retry
+	}
+
+	return nil, false
+}
+
+var errQueueShutdown = errors.New("Queue shutdown while server was not accepting events.")
+
+func (self *HumioQueue) postEvents(ctx context.Context, scope vfilter.Scope,
+				   rows []*ordereddict.Dict) error {
+	nRows := len(rows)
+	opts := vql_subsystem.EncOptsFromScope(scope)
+
+	self.Debug(scope, "posting %v events", nRows)
+
+	// Can happen during plugin shutdown
+	if nRows == 0 {
+		return nil
+	}
+
+	clock := utils.RealClock{}
```

Comment:
utils.GetTime() will allow this to be mocked in tests

Reply:
Ok. I copied this out of another test case. Should the others be updated as well?

Reply:
Same thing, not in 0.6.7-5.